kernel/Makefile

```diff
@@ -93,48 +99,42 @@ KERNEL_VERSIONS+=$(1)
 endif
 
 build_$(2)$(3)$(4): Dockerfile Makefile $(wildcard patches-$(2)/*) $(wildcard config-$(2)*) config-dbg
-	docker pull $(ORG)/$(IMAGE):$(1)$(3)$(4)-$(TAG)$(SUFFIX) || \
-		docker build \
-			--build-arg KERNEL_VERSION=$(1) \
-			--build-arg KERNEL_SERIES=$(2) \
-			--build-arg EXTRA=$(3) \
-			--build-arg DEBUG=$(4) \
-			$(LABELS) \
-			--no-cache -t $(ORG)/$(IMAGE):$(1)$(3)$(4)-$(TAG)$(SUFFIX) .
+	docker pull $(ORG)/$(IMAGE):$(1)$(3)$(4)-$(TAG)$(SUFFIX) || $(MAKE) forcebuild_$(2)$(2)$(4)
```

Comment:
I'm a fan of this pattern of calling make from a make file. It looks very much like an antipattern.

Reply:
I assume you meant that you are _not_ a fan of this pattern?

It isn't an anit-pattern. It actually is quite common, which is why there is a `$(MAKE)` special variable. As one example, it appears ~45 times in the root `Makefile` for the linux kernel [here](https://github.com/torvalds/linux/blob/master/Makefile)

Reply:
Any other comments on it?

Reply:
By the way, even if `$(MAKE)` _were_ an anti-pattern, I would encourage it here. Small anti-patterns are better than wholesale code repetition. But in any case, it isn't an anti-pattern; it is the correct way to call other make targets when they are conditional dependencies.

Reply:
I'm not a fan. most calls to `$(MAKE)` in the Linux kernel top level Make file are with either `-f` to call with another Makefile or `-C` to call `make` in another directory. This is definitely different to call `make` on the same `Makefile` again with another target. That is the part I don't like. The more native `make` way would be to create (temporary) files or just live with a little repetition (which is what I chose). These things don't change much so it's ok

Reply:
I disagree, but I honestly do not care enough about this. Most of the time, I don't touch the kernel or its Makefile. I can revert that part and we can get the rest in.

Reply:
Updated; just the other simplifications remain. Of course, also rebased to latest `master`. Take a look and let us get it through @rn